test/mocks/http/mocks.h

```diff
@@ -195,6 +197,19 @@ class MockStreamDecoderFilterCallbacks : public StreamDecoderFilterCallbacks,
   MOCK_METHOD0(decoderBufferLimit, uint32_t());
 
   // Http::StreamDecoderFilterCallbacks
+  void sendLocalReply(Code code, const std::string& body,
+                      std::function<void(HeaderMap& headers)> modify_headers) override {
+    Utility::sendLocalReply(
+        is_grpc_request_,
+        [this, modify_headers](HeaderMapPtr&& headers, bool end_stream) -> void {
+          if (headers != nullptr && modify_headers != nullptr) {
```

Comment:
headers can't be nullptr?

Reply:
Ditto. Also added the TODO comment you asked for above (GitHub would not let me comment up there).